src/Pos/Block/Logic/VAR.hs

```diff
@@ -203,16 +204,26 @@ applyBlocks calculateLrc pModifier blunds = do
                spanSafe ((==) `on` view (_1 . epochIndexL)) .
                getOldestFirst
 
+data RollbackException = RollbackTipMismatch HeaderHash HeaderHash
```

Comment:
I think it would be better to move this type somewhere. This module contains actual logic. For instance, we have `Pos.Block.Error` module.